app/models/concerns/hydranorth/generic_file/doi_states.rb

```diff
@@ -0,0 +1,90 @@
+module Hydranorth
+  module GenericFile
+    module DOIStates
+      extend ActiveSupport::Concern
+
+      included do
+        after_save :handle_doi_states
+        before_destroy :withdraw_doi
+
+        include AASM
+
+        aasm do
+          state :unpublished, initial: true
+          state :unminted
+          state :excluded
+          state :available
+          state :unsynced
+
+          event :created, after: :queue_create_job do
+            transitions from: :unpublished, to: :unminted
+          end
+
+          event :removed do
+            transitions from: :unpublished, to: :excluded
+          end
+
+          event :unpublish do
+            transitions from: [:excluded, :unsynced], to: :unpublished
+          end
+
+          event :synced do
+            transitions from: [:unminted, :unsynced], to: :available
+          end
+
+          event :altered, after: :queue_update_job do
+            transitions from: [:available, :unpublished], to: :unsynced
+          end
+        end
+
+        def doi_permanent_url
+          "https://doi.org/#{doi.gsub(/doi:/, '')}" if doi.present?
+        end
+
+        def doi_fields_present?
+          # TODO: Shouldn't have to do this as these are required fields on the UI.
+          # However since no data integrity a GF without these fields is technically valid... have to double check
+          self && title.present? && creator.present? && resource_type.present? && Sufia.config.admin_resource_types[resource_type.first].present?
```

Comment:
As comment said, this shouldn't be necessary. But seems like we have no integrity around the data for a GF. Technically the only required field from the back end is a depositor must have edit access (depositor/edit_users fields cant be nil)/ However, from the UI , the required fields are the following: title, creator, resource type, language, subject/keyword, license, and community.  But the only thing validating that these required fields are being set is an html5 required attribute on the fields (and not all browsers support html5 required attributes)! So because of this, I am double checking that the fields required for the DOI service are actually present on the specific GF.